File: pset6/caesar.py
import cs50
import sys

def main():
    if len(sys.argv)!=2 or int(sys.argv[1])<=0:
        print("Please use a non-negative integer as a key")
        exit(1)
    k=int(sys.argv[1])
    print("plaintext: ", end="")
    plaintext=cs50.get_string()
    if plaintext!= None:
        ciphertext=encrypt(plaintext, k)
        print("ciphertext= {}".format(ciphertext))



def encrypt(text, key):
    ciphertext=""
    for i in text:
        if i.isalpha() and i.isupper():
            c= ord(i)
            c= (c-65+key)%26;
            c+=65;
            ciphertext += chr(c)

        elif i.isalpha() and i.islower():
            c= ord(i)
            c= (c-97+key)%26;
            c+=97;
            i= chr(c)
            ciphertext += chr(c)

        else:
            ciphertext += i

    return ciphertext

if __name__ == "__main__":
    main()

File: pset2/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <ctype.h>

int key, i;
string plaintext;
void encrypt()
{
        for (i=0; i<strlen(plaintext); i++)
    {
        if (isalpha(plaintext[i]) && isupper(plaintext[i]))
        {
            int c= (int) plaintext[i];
            c= (c-65+key)%26;
            c+=65;
            plaintext[i]= (char) c;
        }
        else if (isalpha(plaintext[i]) && islower(plaintext[i]))
        {
            int c= (int) plaintext[i];
            c= (c-97+key)%26;
            c+=97;
            plaintext[i]= (char) c;
        }
    }
}
int main(int argc, string argv[])
{
    if (argc==2)
    {
        key=atoi(argv[1]);
        printf("plaintext: ");
        plaintext=get_string();
        if (plaintext!=NULL)
            {
            encrypt();
            printf("ciphertext: %s\n", plaintext);
            }

    }
    else
    {
        printf("Usage: ./caesar k\n");
        return 1;
    }
}